ruma-federation-api/src/identification/identifier_bound/v1.rs

```diff
@@ -0,0 +1,80 @@
+//! [PUT /_matrix/federation/v1/3pid/onbind](https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v1-3pid-onbind)
+
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::{RoomId, ServerKeyId, ServerNameBox, UserId};
+use serde::{Deserialize, Serialize};
+
+use std::collections::BTreeMap;
+
+ruma_api! {
+    metadata: {
+        description: "Used by identity servers to notify the homeserver that one of its users has bound a third party identifier successfully",
+        method: PUT,
+        name: "identifier_bound",
+        path: "/_matrix/federation/v1/3pid/onbind",
+        rate_limited: false,
+        requires_authentication: false,
```

Comment:
We recently made this into an enum, so `false` => `None` and `true` => `AccessToken`. (There's a third variant for federation authentication too, `ServerSignatures`.)

```suggestion
        requires_authentication: None,
```

Reply:
It's also been renamed to just `authentication`.